drivers/modem/modem_shell.c

```diff
@@ -242,11 +245,114 @@ static int cmd_modem_info(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_modem_sms_send(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct ms_context *ms_ctx;
+	struct sms_out sms;
+	int ret= -1;
+
+	if (argc != 4) {
+		shell_fprintf(shell, SHELL_ERROR,
+			      "usage: modem sms send <modem index> <phone number> <message>\n");
+	}
+	else {
+		int modem_index = atoi(argv[1]);
+		ms_ctx = ms_context_from_id(modem_index);
+		if (ms_ctx == NULL) {
+			shell_fprintf(shell, SHELL_ERROR,
+				      "Couldn't open context for modem index %d\n", modem_index);
+			return -1;
+		}
+
+		if (ms_ctx->send_sms == NULL) {
+			shell_fprintf(shell, SHELL_ERROR,
+				      "No function defined to send SMS\n");
+			return -1;
+		}
+
+		snprintf(sms.phone, sizeof(sms.phone), "%s", argv[2]);
+		snprintf(sms.msg, sizeof(sms.msg), "%s", argv[3]);
```

Comment:
I propose `snprintk` instead, then we do not need to include stdio.h and I recall that the snprintk will also use less memory